cmd/config/internal/commands/run_test.go

```diff
@@ -152,6 +152,53 @@ metadata:
 data: {g: h, i: j=k}
 kind: Foo
 apiVersion: v1
+`,
+		},
+		{
+			name: "star",
+			args: []string{"run", "dir",
+				"--enable-star",
+				"--star-path", "a/b/c",
+				"--star-name", "foo",
+				"--", "Foo", "g=h"},
+			path: "dir",
+			expected: `
+metadata:
+  name: function-input
+  annotations:
+    config.kubernetes.io/function: |
+      starlark: {path: a/b/c, name: foo}
+data: {g: h}
+kind: Foo
+apiVersion: v1
+`,
+		},
+		{
+			name: "star-not-enabled",
+			args: []string{"run", "dir",
+				"--star-path", "a/b/c",
+				"--star-name", "foo",
+				"--", "Foo", "g=h"},
+			path: "dir",
+			err:  "must specify --image",
+		},
+		{
+			name: "image-star-not-enabled",
+			args: []string{"run", "dir",
+				"--image", "some_image",
+				"--star-path", "a/b/c",
+				"--star-name", "foo",
+				"--", "Foo", "g=h"},
+			path: "dir",
+			expected: `
+metadata:
```

Comment:
I think this should throw an error something like "star-path specified but enable-star is not set". I don't think we should silently accept the arguments and not honor them.